src/mon/OSDMonitor.cc

```diff
@@ -2271,7 +2271,8 @@ bool OSDMonitor::preprocess_remove_snaps(MonOpRequestRef op)
   MonSession *session = m->get_session();
   if (!session)
     goto ignore;
-  if (!session->is_capable("osd", MON_CAP_R | MON_CAP_W)) {
+  if (!session->caps.is_capable(g_ceph_context, session->entity_name,
+        "osd", "osd pool rmsnap", {}, true, false, false)) {
```

Comment:
MonCapGrant("mon", MON_CAP_R) is enough to make is_capable(..., true, false, false) return true.  I think we should set parameter op_may_write or parameter op_may_exec to true.